### PR TITLE
Drop support for EOL TFMs

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/3806:
- https://github.com/getsentry/sentry-dotnet/issues/3806

### CODE REVIEW

File: test/Sentry.Log4Net.Tests/Sentry.Log4Net.Tests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net8.0;net7.0;net6.0;net48</TargetFrameworks>
+    <TargetFrameworks>net9.0;net8.0;net48</TargetFrameworks>
```
Comment (bruno-garcia) [inline test/Sentry.Log4Net.Tests/Sentry.Log4Net.Tests.csproj:4]: Do we have a net9.0 target for log4net? not sure it's worth adding. Just 8.0 is fine.
In tests we should match/have a target in test that covers

**Review by bruno-garcia — APPROVED** [review]

**Review by vaind — APPROVED** [review]
lgtm, the integration test needs an update though:

```
/Users/runner/.dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.145/tools/Xamarin.Installer.Common.targets(12,3): error : Failed to locate Android SDK at path `/Users/runner/Library/Android/sdk`. [/Users/runner/work/sentry-dotnet/sentry-dotnet/integration-test/maui-app/maui-app.csproj::TargetFramework=net8.0-android34.0]
    0 Warning(s)
    4 Error(s)

Time Elapsed 00:00:05.12

Describing MAUI
Error: [-] Describe MAUI failed
Message
  RuntimeException: Failed to install android dependencies.
  at <ScriptBlock>, /Users/runner/work/sentry-dotnet/sentry-dotnet/integration-test/cli.Tests.ps1:153
  ```

File: test/Sentry.Serilog.Tests/IntegrationTests.verify.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#if NET6_0_OR_GREATER
+#if NET8_0_OR_GREATER
```
[the remark this answers is not in the document]
Comment (jamescrosswell) [reply test/Sentry.Serilog.Tests/IntegrationTests.verify.cs:1]: Yeah you're right - we just need something/anything that will resolve `netstandard2.0`.

**Comment by jamescrosswell** [comment]
> Failed to install android dependencies

From what I can tell, this is an intermittent failure:

> error : Failed to download Xamarin Repository manifest from https://aka.ms/AndroidManifestFeed/d17-12. System.AggregateException: One or more errors occurred. (The SSL connection could not be established, see inner exception.) [/Users/runner/work/sentry-dotnet/sentry-dotnet/integration-test/maui-app/maui-app.csproj::TargetFramework=net8.0-android34.0]

The `aka.ms` servers don't seem very reliable. On my Windows machine, I often have to run `dotnet workload restore` 3 to 4 times before it succeeds (regularly fails to validate the SSL certs).

**Pull request merged**